app/src/main/java/com/theopensourcefamily/chessclock/ClocksPresenter.kt

```diff
@@ -27,19 +31,33 @@ class ClocksPresenter @Inject constructor(
             interaction
           })
         .scan<ClockState>(
-          ClockState.Stopped(30000, 30000),
+          ClockState.Stopped(whitesTime, blacksTime, canReset = false),
           { previousState: ClockState, lastInteraction: ClocksView.Interaction ->
-            if (previousState.blacksTime == 0L || previousState.whitesTime == 0L) {
-              ClockState.GameOver(previousState.whitesTime, previousState.blacksTime)
-            } else {
-              when (lastInteraction) {
-                ClocksView.Interaction.WhitePressed ->
-                  ClockState.BlackRunning(previousState.whitesTime, previousState.blacksTime - 1)
-                ClocksView.Interaction.BlackPressed ->
-                  ClockState.WhiteRunning(previousState.whitesTime - 1, previousState.blacksTime)
-                ClocksView.Interaction.StopPressed ->
-                  ClockState.Stopped(previousState.whitesTime, previousState.blacksTime)
+            when (lastInteraction) {
+              ClocksView.Interaction.WhitePressed -> {
+                if (previousState.blacksTime == 0L || previousState.whitesTime == 0L) {
+                  ClockState.GameOver(previousState.whitesTime, previousState.blacksTime)
+                } else ClockState.BlackRunning(
+                  previousState.whitesTime,
+                  previousState.blacksTime - 1
+                )
+              }
+              ClocksView.Interaction.BlackPressed -> {
+                if (previousState.blacksTime == 0L || previousState.whitesTime == 0L) {
```

Comment:
Do you think that it's possible to extract in a function the WhitePressed interaction code and use it too in the BlackPressed one? The code the two interactions are very similar

Reply:
yess that's what i pointed out in my own comment 😅 I added the WIP label to the ticket because I just wanted to make it work, now I want to refactor that part before rebasing.

Do you have any ideas on where should that logic go? 
a Metod in the presenter?
an extension function over the state class?

Reply:
Sorry I didn't see this comment! D:
After thinking a while, the simplest thing that could be done is create a private functions that checks and sets the game over state:
One function for this -> if (previousState.blacksTime == 0L || previousState.whitesTime == 0L)
Another function for this -> ClockState.GameOver(previousState.whitesTime, previousState.blacksTime)

This is the only code that are really repeating.

¿Did you think in something better?

Reply:
I'll do that :) and hopefully deliver a PR today